src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -395,14 +462,62 @@ def test_member_get_lbound_expression(fortran_writer):
     ssym = DataSymbol("var", stypedef2)
     sref = StructureReference.create(ssym,
                                      ["grid",
-                                      ("map", [two.copy(), two.copy()])])
-    assert sref.member.member.get_lbound_expression(0) == one
-    assert sref.member.member.get_lbound_expression(1) == two
+                                      ("map", [_TWO.copy(), _TWO.copy()])])
+    assert sref.member.member.get_lbound_expression(0) == _ONE
+    assert sref.member.member.get_lbound_expression(1) == _TWO
+    sref2 = StructureReference.create(
+        ssym, [("subgrids", [_TWO.copy(), _TWO.copy()]),
+               ("map", [_TWO.copy(), _TWO.copy()])])
+    assert sref2.member.get_lbound_expression(1) == _TWO
+    assert sref2.member.member.get_lbound_expression(1) == _TWO
+
+
+def test_member_get_ubound_expression(fortran_writer):
```

Comment:
Again, if you refactor then this test can probably be removed. I would say this is a carrot but since you've already written the test it's not a very good one.

Reply:
Tests now use _get_bound_expression" but all tests test for both "upper" and "lower" argument calls.